OpenRA.Mods.TS/Traits/SupportPowers/FireArmamentPower.cs

```diff
@@ -0,0 +1,220 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.GameRules;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	class FireArmamentPowerInfo : SupportPowerInfo
+	{
+		[Desc("Armament names")]
+		public readonly string ArmamentName = "superweapon";
+
+		[Desc("Amount of time before detonation to remove the beacon")]
+		public readonly int BeaconRemoveAdvance = 25;
+
+		[ActorReference]
+		[Desc("Actor to spawn before firing")]
+		public readonly string CameraActor = null;
+
+		[Desc("Amount of time before firing to spawn the camera")]
+		public readonly int CameraSpawnAdvance = 25;
+
+		[Desc("Amount of time after firing to remove the camera")]
+		public readonly int CameraRemoveDelay = 25;
+
+		public override object Create(ActorInitializer init) { return new FireArmamentPower(init.Self, this); }
+	}
+
+	class FireArmamentPower : SupportPower, ITick, INotifyFiredSalvo
+	{
+		readonly FireArmamentPowerInfo info;
+
+		protected Lazy<IFacing> facing;
+		protected Lazy<Armament> armament;
+
+		bool enabled;
+		int ticks;
+		int estimatedTicks;
+		Target target;
+
+		public FireArmamentPower(Actor self, FireArmamentPowerInfo info)
+			: base(self, info)
+		{
+			armament = Exts.Lazy(() => self.TraitsImplementing<Armament>().FirstOrDefault(t => t.Info.Name == info.ArmamentName));
+			facing = Exts.Lazy(() => self.TraitOrDefault<IFacing>());
+			this.info = info;
+			enabled = false;
+		}
+
+		public override void Activate(Actor self, Order order, SupportPowerManager manager)
+		{
+			base.Activate(self, order, manager);
+
+			if (self.Owner.IsAlliedWith(self.World.RenderPlayer))
+				Game.Sound.Play(Info.LaunchSound);
+			else
+				Game.Sound.Play(Info.IncomingSound);
+
+			target = Target.FromCell(self.World, order.TargetLocation);
+
+			enabled = true;
+
+			// TODO: Estimate the projectile travel time somehow
+			estimatedTicks = armament.Value.FireDelay;
```

Comment:
I thought about doing this via doing a rough estimation in the projectile constructors and then sending that back via an interface - since this is needed to be calculated at runtime, due to `Missile`/`GravityBomb` etc. having acceleration - but wouldn't that affect perf heavily?